colossalai/tensor/dist_spec_mgr.py

```diff
@@ -68,6 +68,22 @@ def _gather(tensor: torch.Tensor, old_dist_spec: _DistSpec) -> torch.Tensor:
         assert len(buffer) == 1
         return buffer[0]
 
+    @staticmethod
+    def _all_to_all(tensor: torch.Tensor, old_dist_spec: _DistSpec, dist_spec: _DistSpec) -> torch.Tensor:
```

Comment:
Please add the necessary sanity check for the tensor device as well as the shape.

Reply:
done